table-example/vite.config.ts

```diff
@@ -0,0 +1,20 @@
+import { defineConfig } from 'vite';
+import react from '@vitejs/plugin-react-swc';
+
+// https://vitejs.dev/config/
+export default defineConfig(({ mode }) => ({
+  build: {
+    minify: false,
+    lib: {
+      entry: './src/index.ts',
+      fileName: () => 'index.js',
+      formats: ['cjs'],
+    },
+    rollupOptions: {
+      external: ['react', 'react-dom'],
```

Comment:
We should really make an example that uses `@deephaven/redux` as well... I'm not 100% sure we have this wired up correctly such that a plugin can use `@deephaven/redux` and get the correct version of the store. Likely need to add it as an external: https://github.com/deephaven/web-client-ui/blob/main/packages/code-studio/src/remote-component.config.js#L13

Reply:
I think you can get the right version of the store from redux directly. Since you don't need the store to ask for the store from the provider, then `@deephaven/redux` doesn't have to be external, but probably should be for total correctness

```ts
import { useStore } from 'react-redux';

function MyComponent() {
  const store = useStore();
}
```

This would work if `react-redux` is external. But to use the `useSelector` hook properly, we probably want to externalize `@deephaven/redux`

Reply:
In the plotly express plugin, I currently just import a connected component (ChartPanel) and that works since redux/react-redux are external in that plugin

Reply:
Right, okay. Just something to note, may want to revisit this if we want people using the `useSelector` hook or other parts of `@deephaven/redux`.

Reply:
`useSelector` should work fine and even `@deephaven/redux` not external would work fine since selectors are just lambda functions. The plugin would run into issues if the internal logic of those selectors changed, but not the signature (or if the signature changed)